docs/user-research-prompts.md

```diff
@@ -0,0 +1,328 @@
+    what they would do in the first instance (documentation, come to see us in
+    person, etc.), how often they read the documentation, if they have
+    attended the MO training course.
+  * General *working preferences*: GNOME theme (dark/light), general window
+    placement & use of tabs, etc.
+  * The nine *GUI components of C&R* [listed in turn]: which users use (most),
+    what they use each one for, for how long at a time, & what their ideal
+    would be for each. If they do not use a GUI component, why they do not.
+  * *Component-specific questions*: e.g. for a general component which buttons
+    & widgets they do or do not use, for gcylc which views they use (most) &
+    again if they don't use one why not, for Cylc Review what information
+    (what log or table, etc.) they tend to view & for what reason & at what
+    points they open a new tab ...
+  * How they *interact* with C&R when GUIs are open i.e. when there is a choice,
+    whether they use the the GUI controls & widgets, or use a command, &/or
+    interact directly with the relevant configuration text file.
+  * *Devices*: which kinds they would like to be able to view the C&R UIs on
+    (looking to find out if they might like a mobile/tablet version without
+    asking that straight which would be a leading question).
+  * Their "*flow*" (path) through any GUI components they use, where for any
+    apparent divergences from an ideal efficient flow, what they did as a
+    work-around.
+  * *General input* e.g. what the user considers: the best, & worst, things
+    about the GUIs, what improvements they think are important or would like
+    to see, etc.
+* Aim to gather evidence/data-based information on the following:
+  * *Command-line usage*: ``history | grep cylc`` & ``history | grep rose``
+  * A reference (or zipped directory copy for an external user) to what the
+    user considers a *"typical" suite* they would work with (so we can e.g.
+    see how it appears under the various GUI components e.g. as graphed,
+    in the Tree View, within 'config-edit' if applicable...)
+  * Their *default configuration* to customise GUI components: i.e. their
+    ``~/.cylc/gcylc.rc`` & ``~/.cylc/gscan.rc`` if they have these defined,
+    else go through all config items possible for these & ask what settings
+    they would prefer.
+  * If internal (i.e. MO), the user's suite listing page on Cylc Review.
+* Record everything! Write as much as possible down even if it seems
+  irrelevant. Information can always be ignored but will be lost (even if
+  confined to memory as that is not a reliable source) if not not written down.
+
+
+### 6. By what means do we extract the information (from Q5)?
+
+#### Key considerations:
+
+* Beware of asking *leading questions*! e.g. "*Do you find the GUIs slow?*".
+  The best way to avoid this is by asking questions that are
+  [direct](https://www.usertesting.com/blog/direct-questions/) i.e. open-ended.
```

Comment:
The links in this proposal are very interesting! Thanks @sadielbartholomew ! Especially useful to me that do not have much experience with UI/UX.

Reply:
Me neither, actually, apart from reading some blog posts & attending the two-day course! There is an awful lot of very helpful material online, but it is a matter of finding what is relevant to us, as it is a whole field in itself.

@kinow: one day I will manage to clean up my notes from the UX course. They are actually up already, but only in the "raw" format as copied up electronically straight from my written notes, & may not make much sense. But if you are happy to look at them in the raw state, they are [here](https://github.com/sadielbartholomew/sadielbartholomew.github.io/blob/master/notes/attended-courses/UX-course-feb19.md).